xcm/src/v2/traits.rs

```diff
@@ -313,3 +313,63 @@ impl SendXcm for Tuple {
 		Err(SendError::CannotReachDestination(destination, message))
 	}
 }
+
+/// The info needed to weight an XCM.
+// TODO: Automate Generation
```

Comment:
you mean a macro that takes `Enum XCM` and generates this?

Reply:
yes. it is all boiler plate, and maintaining it is annoying-ish.